### PR TITLE
Add RFC for source/target process modeling

### PR DESCRIPTION
**Pull request by rw-access**
Related to #678 

Started an RFC to model target processes as `process.target.*`

### CODE REVIEW

File: rfcs/text/0000-target-process.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+## References
+
+<!-- Insert any links appropriate to this RFC in this section. -->
+
+### RFC Pull Requests
+
+<!-- An RFC should link to the PRs for each of it stage advancements. -->
+
+* Stage 0: https://github.com/elastic/ecs/pull/NNN
```
Comment (ebeahan) [inline rfcs/text/0000-target-process.md:109]:
```suggestion
* Stage 0: https://github.com/elastic/ecs/pull/1286
```

**Review by ebeahan — APPROVED** [review]
LGTM for stage 0.

Should we hold off merging until @andrewstucki has a chance to review?

**Comment by ebeahan** [comment]
Also, the failed tests should now be addressed by #1283. I can rebase this branch when I set the date and RFC number at merge time.

**Review by ebeahan — APPROVED** [review]
LGTM 👍 

Will hold off merging until @andrewstucki reviews.

**Review by andrewstucki — DISMISSED** [review]
LGTM -- thanks for putting this together @rw-access

**Pull request merged**